mbed_build/_internal/config/config_modifiers/cumulative.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Accumulate values in Config.<attribute>."""
+import itertools
+from dataclasses import dataclass
+from typing import Any, Callable, List, Literal, cast
+
+# Fix circular dependency problem
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mbed_build._internal.config.config import Config
+
+
+class UnableToBuildCumulativeModifier(Exception):
+    """Raised when data given to builder is not valid for cumulative modifier."""
+
+
+# TODO: potentially add more accumulating properties here.
+# Two projects this has been tested against:
+# - mbed-cloud-client-example
+# - mbed-os-example-blinky
+# only contain "features_add" and "features_remove".
+ACCUMULATING_OVERRIDES = ("features",)
```

Comment:
There is the full list here: https://github.com/ARMmbed/mbed-targets/blob/master/mbed_targets/_internal/targets_json_parsers/accumulating_attribute_parser.py#L15

Reply:
Oh yeah, I had a full list before here, but decided to go only with what we need right now. My thought process was - let's not add things to config we're not 100% sure is necessary right now. I could link to that full list, thoughts?

I'm a bit concerned most of these are `targets.json` specific. 🤔

Reply:
I've added a full list of those attributes.